mkfs/rootdir.c

```diff
@@ -362,6 +362,76 @@ static int add_symbolic_link(struct btrfs_trans_handle *trans,
 	return ret;
 }
 
+/*
+ * keep our extent size at 1MB max, this makes it easier to work
+ * inside the tiny block groups created during mkfs
+ */
+#define MAX_EXTENT_SIZE SZ_1M
+
+static int read_and_write_extent(struct btrfs_trans_handle *trans,
+				 struct btrfs_root *root,
+				 struct btrfs_inode_item *btrfs_inode,
+				 u64 objectid, int fd, u64 file_pos, char *buf,
+				 u64 size, const char *path_name)
+{
+	int ret;
+	u32 sectorsize = root->fs_info->sectorsize;
+	u64 bytes_read, first_block, to_read, to_write;
+	struct btrfs_key key;
+
+	to_read = min(file_pos + MAX_EXTENT_SIZE, size) - file_pos;
+
+	bytes_read = 0;
+
+	while (bytes_read < to_read) {
+		ssize_t ret_read;
+
+		ret_read = pread(fd, buf + bytes_read, to_read - bytes_read,
+				 file_pos + bytes_read);
+		if (ret_read == -1) {
```

Comment:
Please use `ret_read < 0`, as that's the common practice for detecting an error, not only for the common IO but also all the other btrfs specific calls.